### PR TITLE
External instrumenter

### PR DESCRIPTION
**Pull request by Quarub**
Use dead_instrumenter.

Note: Requires version 0.0.2 of dead_instrumenter to be merged and published first! [See here](https://github.com/DeadCodeProductions/dead_instrumenter/pull/2)

### CODE REVIEW

File: checker.py [hunk 1]
```diff
@@ -250,6 +251,11 @@ def sanitize(
         bool: True if nothing indicative of undefined behaviour is found.
     """
     # Taking advantage of shortciruit logic...
+    # a = check_compiler_warnings(gcc, clang, file, flags, cc_timeout)
```
Comment (thetheodor) [inline checker.py:254]: Drop commented out code?

File: checker.py [hunk 2]
```diff
@@ -263,23 +269,23 @@ def sanitize(
...
-        include_paths (list[str]): Include paths to use when compiling.
-    """
-    cmd = [annotator, str(file)]
-    for path in include_paths:
-        cmd.append(f"--extra-arg=-isystem{path}")
-    try:
-        utils.run_cmd(cmd)
-    except subprocess.CalledProcessError as e:
-        raise Exception("Static annotator failed to annotate {file}! {e}")
+# def annotate_program_with_static(
```
Comment (thetheodor) [inline checker.py:272]: Drop commented out code?

File: checker.py [hunk 3]
```diff
@@ -374,12 +380,10 @@ def is_interesting_with_static_globals(self, case: utils.Case) -> bool:
                 print(case.code, file=new_cfile)
 
             # TODO: Handle include_paths better
-            include_paths = utils.find_include_paths(
-                self.config.llvm.sane_version, tf.name, case.bad_setting.get_flag_str()
-            )
-            annotate_program_with_static(
-                self.config.static_annotator, Path(tf.name), include_paths
-            )
+            import pdb
```
Comment (thetheodor) [inline checker.py:383]: Why do we need to use the python debugger here?

File: requirements_hook.txt [hunk 4]
```diff
@@ -1,6 +1,7 @@
 attrs>=21.4.0
```
Comment (thetheodor) [inline requirements_hook.txt:1]: Does it make sense to merge `requirements_hook.txt` and `requirements.txt`?

**Review by thetheodor — CHANGES_REQUESTED** [review]

File: checker.py (hunk 3 above)
Comment (Quarub) [reply checker.py:383]: Whoops; forgot my pre-commit hook :(

Shall we consider something like https://pre-commit.com/ ?

Comment (thetheodor) [reply checker.py:383]: Yup, there's also this: https://github.com/PyCQA/flake8

File: requirements_hook.txt (hunk 4 above)
Comment (Quarub) [reply requirements_hook.txt:1]: I like it that we have minimal dependencies for people that want to install it.

Maybe with a pre-commit hook environment as shown in the comment above we can drop the dev requirements all together?
They just include pytype, mypy and black iirc.

Comment (thetheodor) [reply requirements_hook.txt:1]: Then maybe just rename requirements_hook to requirements-dev?

**Review by thetheodor — APPROVED** [review]

**Pull request merged**